Binary Search/BinarySearch.java

```diff
@@ -0,0 +1,47 @@
+import java.util.ArrayList;
```

Comment:
You should add yourself as a contributor at the beginning of the file (same for the py solution)! What about the time and space complexity? You can follow the example of other files from the repo.

Reply:
I am appreciate for your feedbacks. I will do my best for the next time.